File: app/src/main/java/com/github/willjgriff/ethereumwallet/ui/settings/mvp/ChangeAddressPresenter.java
package com.github.willjgriff.ethereumwallet.ui.settings.mvp;

import com.github.willjgriff.ethereumwallet.data.model.DomainAddress;
import com.github.willjgriff.ethereumwallet.ethereum.account.AddressManager;
import com.github.willjgriff.ethereumwallet.mvp.BaseMvpPresenter;

import javax.inject.Inject;

/**
 * Created by Will on 28/02/2017.
 */

public class ChangeAddressPresenter extends BaseMvpPresenter<ChangeAddressView> {

	private AddressManager mAddressManager;

	@Inject
	public ChangeAddressPresenter(AddressManager addressManager) {
		mAddressManager = addressManager;
	}

	@Override
	public void viewReady() {
		getView().setAddresses(mAddressManager.getAllAddresses());
	}

	public void onAddressItemClicked(DomainAddress domainAddress) {
		mAddressManager.setActiveAccount(domainAddress);
		getView().closeScreen();
	}
}

File: app/src/main/kotlin/com/github/willjgriff/ethereumwallet/ui/screens/settings/mvp/ChangeAddressPresenter.kt
package com.github.willjgriff.ethereumwallet.ui.screens.settings.mvp

import com.github.willjgriff.ethereumwallet.ethereum.address.AddressManager
import com.github.willjgriff.ethereumwallet.ethereum.address.model.DomainAddress
import com.github.willjgriff.ethereumwallet.mvp.BaseMvpPresenterKotlin
import javax.inject.Inject

/**
 * Created by williamgriffiths on 04/05/2017.
 */
class ChangeAddressPresenter @Inject constructor(private val addressManager: AddressManager)
    : BaseMvpPresenterKotlin<ChangeAddressView>() {

    override fun viewReady() {
        view?.setAddresses(addressManager.getAllAddresses())
    }

    fun onAddressItemClicked(domainAddress: DomainAddress) {
        addressManager.setActiveAccount(domainAddress)
        view?.closeScreen()
    }
}
